File: Codes/javase/src/lesson7/Computer.java
package lesson7;

class Computer {

    Integer id;
    String brand;
    String model;
    String color;

    Computer() {
        System.out.println("Default constructor işlədi.");
    }

    Computer(String model) {
        this.model = model;
        System.out.println("Constructor (model): " + model);
    }

    Computer(String model, String color) {
        this.model = model;
        this.color = color;
        System.out.println("Constructor (model, color): " + model + ", " + color);
    }
}

File: codes/spring/frontend/computerShopping/lesson013/assets/computer.js
let form = document.querySelector('form');

form.addEventListener('submit', async (e) => {
    e.preventDefault();

    let computer = {
        brand: document.getElementById('brand').value,
        model: document.getElementById('model').value
    };

    try {
        let response = await fetch('http://localhost:8008/computers/addComputer', {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(computer)
        });

        let data = await response.text();
        console.log(data);

    } catch (error) {
        console.error("Xəta:", error);
    }
});